src/components/form/FileInput/FileInput.tsx

```diff
@@ -1,62 +1,130 @@
 import Popup from 'components/popups/Popup';
 
+export interface FileInputProps {
+    /**
+     * string value to be displayed, creates controlled component if specified
+     */
+    value?: string | null;
+
+    /**
+     * name of the field appended to 'fileName' string
+     */
+    name?: string;
+
+    /**
+     * specifies a short hint that describes the expected input
+     */
+    placeholder?: string;
+
+    /**
+     * function called on file change
+     */
+    onChange?: (fileOrFiles: File | File[] | null, filename?: string) => void;
+
+    /**
+     * function called on file reset
+     */
+    onReset?: () => void;
+
+    /**
+     * if set to true opening file selector will be disabled
+     */
+    loading?: boolean;
+
+    /**
+     * if set to true component will be disabled
+     */
+    disabled?: boolean;
+
+    /**
+     * if set to false input string field will not be presented
+     */
+    showInput?: boolean;
+
+    /**
+     * if set to false reset button will not be presented
+     */
+    showReset?: boolean;
+
+    /**
+     * additional parameters for open file button, props for Button component
+     */
+    openButtonParams?: StrictButtonProps;
+
+    /**
+     * additional help information shown in Popup
+     */
+    help?: string;
+
+    /**
+     * if set to true multiple files can be selected
+     */
+    multiple?: boolean;
+}
+
 /**
  * FileInput is a component showing file input field. Can be displayed as normal input field containing name of the chosen file
  * or as a simple button triggering opening file browser. You can add reset button to unset chosen file.
  *
  * Accessible as `FileInput` or `Form.File`.
  */
-export default function FileInput({
-    value,
-    name,
-    placeholder,
-    onChange,
-    onReset,
-    loading,
-    disabled,
-    showInput,
-    showReset,
-    openButtonParams,
-    help,
-    multiple
-}) {
-    const inputRef = createRef();
+const FileInput: FunctionComponent<FileInputProps> = ({
+    value = null,
+    name = '',
+    placeholder = '',
+    onChange = noop,
+    onReset = noop,
+    loading = false,
+    disabled = false,
+    showInput = true,
+    showReset = true,
+    openButtonParams = {},
+    help = '',
+    multiple = false
+}) => {
+    const inputRef = createRef<HTMLInputElement>();
     const [internalValue, setInternalValue] = useState('');
 
-    const openFileSelection = e => {
-        e.preventDefault();
-        inputRef.current.click();
-        return false;
+    const openFileSelection = (event: React.MouseEvent<HTMLButtonElement>, _data: ButtonProps) => {
+        event.preventDefault();
+        inputRef?.current?.click();
     };
 
     const resetInput = () => {
         setInternalValue('');
         onChange(null, '');
     };
 
-    const resetFileSelection = e => {
-        e.preventDefault();
+    const resetFileSelection = (event: React.MouseEvent<HTMLButtonElement>, _data: ButtonProps) => {
```

Comment:
WDYT about the below simplification? 🤔 

```suggestion
    const resetFileSelection: ButtonProps['onClick'] = event => {
```

Reply:
Done.